sdk/openai/azure-ai-openai-assistants/src/main/java/com/azure/ai/openai/assistants/AssistantsAsyncClient.java

```diff
@@ -2355,6 +2359,31 @@ public Mono<ThreadRun> createThreadAndRun(CreateAndRunThreadOptions createAndRun
             .map(protocolMethodData -> protocolMethodData.toObject(ThreadRun.class));
     }
 
+    /**
+     * Creates a new assistant thread and immediately starts a run using that new thread. Updates are returned as a
+     * stream.
+     *
+     * @param createAndRunThreadOptions The details used when creating and immediately running a new assistant thread.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return data representing a single evaluation run of an assistant thread on successful completion of
+     * {@link Flux}.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public Flux<StreamUpdate> createThreadAndRunStream(CreateAndRunThreadOptions createAndRunThreadOptions) {
```

Comment:
Why can't we return `Flux<StreamThreadCreation>` instead of `Flux<StreamUpdate>`? Users will have to cast this on every single even if we just return the empty abstract base class.

Reply:
This method returns many different subtypes of `StreamUpdate`. `StreamThreadCreation` is only one of them. I went into more details in this answer: https://github.com/Azure/azure-sdk-for-java/pull/39733#discussion_r1592693268